rest/config_legacy_test.go

```diff
@@ -32,8 +32,8 @@ func TestLegacyConfigToStartupConfig(t *testing.T) {
 		{
 			name:     "Override duration zero ServerReadTimeout",
 			base:     StartupConfig{API: APIConfig{ServerReadTimeout: base.NewConfigDuration(time.Second * 10)}},
-			input:    LegacyServerConfig{ServerReadTimeout: base.IntPtr(0)},
-			expected: StartupConfig{API: APIConfig{ServerReadTimeout: base.NewConfigDuration(0)}},
+			input:    LegacyServerConfig{ServerReadTimeout: base.IntPtr(1)},
+			expected: StartupConfig{API: APIConfig{ServerReadTimeout: base.NewConfigDuration(time.Second)}},
```

Comment:
Doesn't match test case name any more - but looking at the test-case again it looks like we'd want a pointer to `0` to override the original value, as it's explicitly set?

I can't remember why this stopped working. Is it related to `D` inside `ConfigDuration` not being a pointer type?